WooCommerce/Classes/ViewRelated/Dashboard/Settings/In-Person Payments/AboutTapToPayViewModel.swift

```diff
@@ -16,19 +16,50 @@ class AboutTapToPayViewModel: ObservableObject {
             authenticated: false)
     }()
 
+    let siteID: Int64
+
     let formattedMinimumOperatingSystemVersionForTapToPay: String
 
-    init(configuration: CardPresentPaymentsConfiguration,
-         buttonAction: (() -> Void)?) {
+    let cardPresentPaymentsOnboardingUseCase: CardPresentPaymentsOnboardingUseCase
+
+    let shouldAlwaysHideSetUpTapToPayButton: Bool
+
+    init(siteID: Int64 = ServiceLocator.stores.sessionManager.defaultStoreID ?? 0,
+         configuration: CardPresentPaymentsConfiguration = CardPresentConfigurationLoader().configuration,
+         cardReaderSupportDeterminer: CardReaderSupportDetermining? = nil,
+         cardPresentPaymentsOnboardingUseCase: CardPresentPaymentsOnboardingUseCase? = nil,
+         shouldAlwaysHideSetUpTapToPayButton: Bool = false) {
+        self.siteID = siteID
         self.configuration = configuration
-        self.buttonAction = buttonAction
-        shouldShowButton = buttonAction != nil
+        self.cardReaderSupportDeterminer = cardReaderSupportDeterminer ?? CardReaderSupportDeterminer(siteID: siteID, configuration: configuration)
+
+        if let cardPresentPaymentsOnboardingUseCase {
+            self.cardPresentPaymentsOnboardingUseCase = cardPresentPaymentsOnboardingUseCase
+        } else {
```

Comment:
Just a thought, I'm wondering if we could pass always the `CardPresentPaymentsOnboardingUseCase` dependency from the outside, so it never has to create its own dependency. I would understand if it's a hassle though 👍

Reply:
I looked at doing that – it's a hassle from the Set Up TTP Learn More links, because those screens don't have it. The onboarding isn't even used when the About screen is opened from there, as there is no button, so we could just make it `nil`, but that would mean it's broken if someone else ever uses it with that init method.